tests/test_service_injection.py

```diff
@@ -0,0 +1,196 @@
+import copy
+from configparser import ConfigParser
+from itertools import permutations
+from sys import implementation
+from types import MethodType
+
+from kink import inject
+import pytest
+
+from app.server import create_app
+from app.services import NopStore, interfaces
+from app.services.inventory import INVENTORY
+from app.services.container import Injector
+from app.services.container import ProtocolError, UnknownImplementationError
+
+
+nop_config = ConfigParser()
+nop_config.add_section("STORE")
+nop_config["STORE"]["default_implementation"] = "Nop"
+nop_config.add_section("MESSENGER")
+nop_config["MESSENGER"]["default_implementation"] = "Nop"
+
+
+def test_documentation_example():
+    """
+    Sanity check that the documented test example works
+    """
+
+    test_store = NopStore
+    test_store.get_record = MethodType(lambda x: {"mock": "record"}, test_store)
```

Comment:
Do we need to reset the `NopStore` to its original state at the end of each test?

Reply:
really good question, missed the implication, I will check.

Reply:
good spot, we'd need to reset it, which is nasty. I'll add a switch instead so we can situationally flag off the type check and pass in a mock instead.

Reply:
We could just define a distinct mock implementation for each test? I don't see that we need to be able to pass in stuff which doesn't pass the type-check.